cmd/dep/init.go

```diff
@@ -209,6 +227,45 @@ func hasImportPathPrefix(s, prefix string) bool {
 	return strings.HasPrefix(s, prefix+"/")
 }
 
+// getVersionConstituents extracts version constituents
+func getVersionConstituents(v gps.Version) (version, revision string) {
+	switch tv := v.(type) {
+	case gps.UnpairedVersion:
+		version = tv.String()
+	case gps.Revision:
+		revision = tv.String()
+	case gps.PairedVersion:
+		version = tv.Unpair().String()
+		revision = tv.Underlying().String()
+	}
+
+	return version, revision
+}
+
+// getProjectPropertiesFromVersion takes a gps.Version and returns a proper
+// gps.ProjectProperties with Constraint value based on the provided version.
+func getProjectPropertiesFromVersion(v gps.Version) gps.ProjectProperties {
+	pp := gps.ProjectProperties{}
+
+	// constituent version and revision. Ignoring revison for manifest.
+	cv, _ := getVersionConstituents(v)
```

Comment:
Rather than converting to and from strings, we can do it all with the types - in `getProjectPropertiesFromVersion`, we can do something like this:

```go
switch tv := v.(type) {
case gps.PairedVersion:
	v = tv.Unpair()
case gps.Revision:
	return pp
}
```

Now the `v` is exactly what we'll want to assign into `pp.Constraint`, unless we're in the semver case.